lib/restricted-input.js

```diff
@@ -76,105 +73,63 @@ RestrictedInput.prototype._attachEvents = function () {
   this.inputElement.addEventListener('keydown', function (event) {
     if (keyCannotMutateValue(event)) { return; }
     if (self._isDeletion(event)) {
-      self._unformatInput(event);
+      self._stateToFormat = self.formatter.simulateDeletion({
+        event: event,
+        selection: getSelection(self.inputElement),
+        value: self.inputElement.value
+      });
     }
   });
-  this.inputElement.addEventListener('keypress', function (event) {
-    if (keyCannotMutateValue(event)) { return; }
-    self._unformatInput(event);
-  });
-  this.inputElement.addEventListener('keyup', function (event) {
-    self._reformatInput(event);
-  });
-  this.inputElement.addEventListener('input', function (event) {
-    self._reformatInput(event);
-  });
-  this.inputElement.addEventListener('paste', function (event) {
-    self._handlePasteEvent(event);
-  });
+
+  // TODO: improve this isIE9 logic
+  if (IE_9_REGEX.test(navigator.userAgent)) {
+    this.inputElement.addEventListener('keyup', this._reformatInput.bind(this));
+  } else {
+    this.inputElement.addEventListener('input', this._reformatInput.bind(this));
+  }
+
+  this.inputElement.addEventListener('focus', this._reformatInput.bind(this));
```

Comment:
This is good for if there's a permacharacter at the beginning of an input. IE, if the pattern is `--{{9}}z{{a}}b`, then focussing the input will auto add the `--`.